File: examples/quickstart.py
"""Load a tokenizer and encode/decode text with the gigatoken API.

Run with: uv run examples/quickstart.py
"""

import awkward as ak

import gigatoken as gt

# Accepts a HuggingFace Hub repo id, a path to a tokenizer.json (or a
# directory containing one), or an already-initialized HF tokenizer.
tokenizer = gt.Tokenizer("openai-community/gpt2")
print(f"loaded {tokenizer!r} with vocab_size={tokenizer.vocab_size}")

# Single documents: encode returns a numpy uint32 array, decode returns bytes.
ids = tokenizer.encode("Tokenize your text data at GB/s!")
print(ids)
print(tokenizer.decode(ids).decode("utf-8"))

# Batches encode in parallel and return a ragged awkward Array,
# one row of token ids per document.
docs = ["The first document.", "And a second, slightly longer one."]
tokens = tokenizer.encode_batch(docs)
print(tokens)
print(ak.num(tokens))  # tokens per document

# For model input, assemble a padded (rows x width) matrix directly in Rust,
# along with each row's real length.
matrix, lengths = tokenizer.encode_batch_padded(docs, pad_id=0)
print(matrix)
print(lengths)

File: examples/quickstart.js
/**
 * Quickstart example for GigaToken Node.js / TypeScript API
 *
 * Run with: node examples/quickstart.js
 */

const { GigaTokenizer } = require('../index')
const path = require('path')

// 1. Load tokenizer from a tiktoken rank file
const rankPath = path.join(__dirname, '../cl100k_base.tiktoken')
const tokenizer = GigaTokenizer.fromTiktoken(rankPath)

console.log('--- 1. Single Document Token Counting & Encoding ---')
const sampleText = 'Tokenize your text data at GB/s with Neosantara AI!'
const count = tokenizer.countTokens(sampleText)
const tokenIds = tokenizer.encode(sampleText)

console.log('Sample Text  :', sampleText)
console.log('Token Count  :', count)
console.log('Token IDs    :', tokenIds)

console.log('\n--- 2. Batch Document Token Counting (Rayon Parallel SIMD) ---')
const docs = [
  'Document 1: Halo Neosantara AI Gateway',
  'Document 2: High performance tokenization with GigaToken NAPI',
  'Document 3: Pemrosesan data bahasa Indonesia super cepat'
]

const batchCounts = tokenizer.countTokensBatch(docs)
console.log('Batch Token Counts per Doc:', batchCounts)
